scripts/variantstore/wdl/extract/workflow_compute_costs.py

```diff
@@ -0,0 +1,129 @@
+from logging import info, warning
+import urllib.parse
+
+
+def fapi_list_submissions(workspace_namespace: str, workspace_name: str):
+    from firecloud import api as fapi
+    return fapi.list_submissions(workspace_namespace, workspace_name).json()
+
+
+def fapi_get_submission(workspace_namespace: str, workspace_name: str, submission_id: str):
+    from firecloud import api as fapi
+    return fapi.get_submission(workspace_namespace, workspace_name, submission_id).json()
+
+
+def fapi_get_workflow_metadata(workspace_namespace: str, workspace_name: str, submission_id: str, workflow_id: str):
+    from firecloud import api as fapi
+    return fapi.get_workflow_metadata(workspace_namespace, workspace_name, submission_id, workflow_id).json()
+
+
+def compute_costs(workspace_namespace, workspace_name, excluded_submission_ids,
+                  list_submissions=fapi_list_submissions,
+                  get_submission=fapi_get_submission,
+                  get_workflow_metadata=fapi_get_workflow_metadata):
+
+    submissions = list_submissions(workspace_namespace, workspace_name)
+    submission_ids = [s['submissionId'] for s in submissions]
+
+    from collections import defaultdict
+    workflow_costs = defaultdict(dict)
+
+    for submission_id in submission_ids:
+        if excluded_submission_ids and submission_id in excluded_submission_ids:
+            info(f"Submission id '{submission_id}' in exclude list, skipping cost calculation.")
+            continue
+        submission = get_submission(workspace_namespace, workspace_name, submission_id)
+        workflows = submission['workflows']
+        if not workflows:
+            warning(f"Submission '{submission_id}' has no workflows, skipping cost calculation.")
+            continue
+        workflow_ids = [w['workflowId'] for w in workflows]
+        for workflow_id in workflow_ids:
+            workflow = get_workflow_metadata(workspace_namespace, workspace_name, submission_id, workflow_id)
+            workflow_name = workflow['workflowName']
+            if not workflow_name:
+                warning(f"Workflow {workflow_id} has no workflow name, skipping cost calculation.")
+                continue
+
+            if workflow_name.startswith('Gvs'):
+                if len(workflow_ids) == 1:
+                    # If this run is < 1 day old the cost data may not yet be available.
```

Comment:
Is it that the cost data might not be available - or is it that it might be not yet up-to-date? Perhaps put a general warning if the run is < 1 day old to tell the user that they *might* want to run this later just to be sure that cost is correct?

Reply:
I think this usually happens when this code runs too soon after the workflow completes. 

I am capturing this message in a logfile but now that you bring this up I'm wondering if that's really a place that the users of this code would actually look; we should probably discuss as a team the best way to represent this condition so users can make an informed choice as to whether instances of this situation are okay or not.

Reply:
per mob discussion we're going to use this as is for now